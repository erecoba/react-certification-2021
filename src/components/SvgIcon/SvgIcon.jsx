import React from 'react';

import { SvgWrapper } from './SvgIcon.styled';

const SvgIcon = ({
  children,
  color,
  viewBox = '0 0 24 24',
  focusable = false,
  size = '1.5rem',
  ...rest
}) => {
  return (
    <SvgWrapper
      viewBox={viewBox}
      focusable={focusable}
      size={size}
      {...rest}
      color={color}
    >
      {children}
    </SvgWrapper>
  );
};

export default SvgIcon;
