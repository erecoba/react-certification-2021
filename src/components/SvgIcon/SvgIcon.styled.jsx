import styled from 'styled-components';

const SvgWrapper = styled.svg`
  display: inline-block;
  color: ${({ color, theme }) => (color ? theme.colors[color] : theme.label)};
  width: ${(props) => props.size};
  height: ${(props) => props.size};
  fill: ${({ color, theme }) => (color ? theme.colors[color] : theme.label)};
  font-size: 1.5rem;
  user-select: none;
  flex-shrink: 0;
`;

export { SvgWrapper };
