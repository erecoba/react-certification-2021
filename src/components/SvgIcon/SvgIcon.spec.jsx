import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { ThemeProvider } from 'styled-components';
import SvgIcon from './SvgIcon';
import { SvgWrapper } from './SvgIcon.styled';

describe('SvgIcon', () => {
  it('should render correctly with svg', () => {
    const { container } = render(
      <SvgIcon>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    );

    expect(container.tagName).toMatchSnapshot('svg');
  });

  it('should render with a color', () => {
    const { container } = render(
      <ThemeProvider theme={{ colors: { base: '#f0f' } }}>
        <SvgWrapper color="base" />
      </ThemeProvider>
    );
    expect(container.firstChild).toHaveStyleRule('fill', '#f0f');
  });
});
