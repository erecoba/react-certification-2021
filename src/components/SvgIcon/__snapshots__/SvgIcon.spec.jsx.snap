// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon should render correctly with svg: svg 1`] = `"DIV"`;
